client/src/app/core/actions/committee-action.ts

```diff
@@ -1,40 +1,46 @@
 import { Identifiable } from 'app/shared/models/base/identifiable';
+import { DataClass } from '../decorators/data-class';
 import { Id, UnsafeHtml } from '../definitions/key-types';
 
-export namespace CommitteeAction {
-    export const CREATE = 'committee.create';
-    export const UPDATE = 'committee.update';
-    export const DELETE = 'committee.delete';
+export class CommitteeAction {
+    public static readonly CREATE = 'committee.create';
+    public static readonly UPDATE = 'committee.update';
+    public static readonly DELETE = 'committee.delete';
+}
 
-    export interface PartialPayload {
-        description?: UnsafeHtml;
-        // For UPDATE: Required permission: OML.can_manage_organization
-        user_ids?: Id[];
-        manager_ids?: Id[];
-        // For UPDATE: Required permission: CML.can_manage || OML.can_manage_organization
-        organization_tag_ids?: Id[];
-        // Required permission: OML.can_manage_organization
-        forward_to_committee_ids?: Id[];
-        receive_forwardings_from_committee_ids?: Id[];
-    }
+class CommitteePartialPayload {
+    description?: UnsafeHtml = '';
+    // For UPDATE: Required permission: OML.can_manage_organization
+    user_ids?: Id[] = [];
+    manager_ids?: Id[] = [];
+    // For UPDATE: Required permission: CML.can_manage || OML.can_manage_organization
+    organization_tag_ids?: Id[] = [];
+    // Required permission: OML.can_manage_organization
+    forward_to_committee_ids?: Id[] = [];
+    receive_forwardings_from_committee_ids?: Id[] = [];
```

Comment:
We could here also use decorators to inject default values, but I think, that will be an extra expense but has no additional value...

Reply:
I do not like the defaults. As I can see the actual class is used in data-class.ts line 32 ff.:
```
for (const key in dataClass) {
    result[key] = value[key];
}
```
1. You do nat care about types
2. You do not care about the default values
3. you just need the field names.

Why not just use a list of fields to copy:
```
CommitteePartialPayloadFields = ["description", "user_ids", "manager_ids", ...]
```
This contains all information you need without the unnecessary defaults (and types)

Reply:
I know, that there are then unnecessary default values and types. My reason for it was, that we can "combine" definitions (like an interface) and such fields to copy.
If we use a separate list of fields to indicate which fields have to be "copied", we have two places to maintain - the definitions as well as the list of fields. 
I can use the default values to check if the copied values have the correct types.

Reply:
> I can use the default values to check if the copied values have the correct types.

Reply:
> Please add an example then, so we can see the usecase.
>

See the file `committee-repository.service.ts` line 65 to 69: There I replaced the "old" `CommitteeAction.UpdatePayload` with the "new" class `CommitteeUpdatePayload`. 
 
> Currently, there are also two places: In the actions and the model fields.
> 

Well, the models and actions do not contain the same keys.

> I would suggest to add the list of fields into the model classes (e.g. motion.ts) and add a constants to every class (like `FIELDS`). Yes, we have them twice, but then in the same file. But I guess you can use typescript's information to extract every key from the interface. A quick google search: https://stackoverflow.com/questions/43909566/get-keys-of-a-typescript-interface-as-array-of-strings and https://github.com/kimamula/ts-transformer-keys
> 
> We have all fields in each model file (There we need them explicitly once). Based on this list you can add/remove additional fields and build up the list of fields for each action.

The problem is, that `const`s cannot be decorated (as mentioned in the comment above).